src/Database/Migrate.php

```
@@ -200,19 +203,36 @@ protected function preSyncActions()
-            } else {
-                $this->tableHandler->addIndex($key, $tableName, $keyData['columns'], $keyData['unique']);
+        if (isset($targetTable['keys']) && is_array($targetTable['keys'])) {
+            foreach ($targetTable['keys'] as $key => $keyData) {
+                if (!is_string($key)) {
+                    continue;
+                }
+                if ($key === 'PRIMARY') {
+                    $this->tableHandler->addPrimaryKey($tableName, $keyData['columns']);
+                } else {
+                    $this->tableHandler->addIndex($key, $tableName, $keyData['columns'], $keyData['unique']);
+                }
```

PHPMD (2.15.0)

[error] The method addMissingTable uses an else expression. Else clauses are basically not necessary and you can simplify the code by not using them. (undefined)

(ElseExpression)